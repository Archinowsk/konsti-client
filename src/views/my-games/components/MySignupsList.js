/* @flow */
import React from 'react'
import { withNamespaces } from 'react-i18next'
import { Link } from 'react-router-dom'
import moment from 'moment'
import timeFormatter from 'utils/timeFormatter'

type Props = {
  t: Function,
  signedGames: Array<any>,
}

const MySignupsList = (props: Props) => {
  const { signedGames, t } = props

  // Sort games by time and name
  const sortedGames = signedGames.sort((a, b) => {
    const keyA = moment(a.details.startTime) + a.details.title.toLowerCase()
    const keyB = moment(b.details.startTime) + b.details.title.toLowerCase()
    if (keyA < keyB) return -1
    if (keyA > keyB) return 1
    return 0
  })

  const GamesList = sortedGames.map(game => {
    const formattedDate = timeFormatter.weekdayAndTime(game.details.startTime)

    return (
      <li key={game.id}>
        <Link to={`/games/${game.id}`}>
          {formattedDate}: {game.details.title} ({game.priority})
        </Link>
      </li>
    )
  })

  return (
    <div className="my-signups-list">
      <p>{t('signedGames')}</p>
      <ul>
        {signedGames.length === 0 && <span>{t('noSignedGames')}</span>}
        {GamesList}
      </ul>
    </div>
  )
}

export default withNamespaces()(MySignupsList)
