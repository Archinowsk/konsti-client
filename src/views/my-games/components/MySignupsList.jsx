import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { Link } from 'react-router-dom';
import moment from 'moment';

const MySignupsList = props => {
  const { signedGames, t } = props;

  // Sort games by time and name
  const sortedGames = signedGames.sort((a, b) => {
    const keyA = moment(a.date) + a.title;
    const keyB = moment(b.date) + b.title;
    if (keyA < keyB) return -1;
    if (keyA > keyB) return 1;
    return 0;
  });

  const GamesList = signedGames.map(game => {
    const formattedDate = moment.utc(game.date).format('DD.M.YYYY HH:mm');

    return (
      <li key={game.id}>
        <Link to={`/games/${game.id}`}>
          {formattedDate}: {game.title} ({game.priority})
        </Link>
      </li>
    );
  });

  return (
    <div>
      <p>
        {t('signedGames')}
      </p>
      <ul>
        {signedGames.length === 0 &&
          <span>
            {t('noSignedGames')}
          </span>}
        {GamesList}
      </ul>
    </div>
  );
};

MySignupsList.propTypes = {
  t: PropTypes.func.isRequired,
  signedGames: PropTypes.array.isRequired,
};

export default translate()(MySignupsList);
