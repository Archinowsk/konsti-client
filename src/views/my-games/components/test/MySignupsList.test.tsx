import React from 'react';
import { shallow } from 'enzyme';
import { MySignupsList, Props } from '../MySignupsList';

const signedGames = [];

describe('MySignupsList', () => {
  it('should render correctly', () => {
    const props: Props = { signedGames };
    const component = shallow(<MySignupsList {...props} />);
    expect(component).toMatchSnapshot();
  });
});
