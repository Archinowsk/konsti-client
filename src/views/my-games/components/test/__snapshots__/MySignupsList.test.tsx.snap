// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySignupsList should render correctly 1`] = `
<div
  className="my-signups-list"
>
  <h3>
    signedGames
  </h3>
  <div
    className="my-signups-games"
  >
    <span>
      noSignedGames
    </span>
  </div>
</div>
`;
